docs/features/assets.md

```diff
@@ -1,52 +1,178 @@
+npx @alleyinteractive/create-block
+```
+
+Without providing any options the tool will prompt the user through several
+options for creating a block.
+
+Once the block is created, the block will be automatically registered and loaded
+by Mantle via the `Asset_Service_Provider`'s `load_blocks` method.
 
 ## Registering Assets
 
-Assets can be enqueued in a fluent-basis on top of the existing WordPress API:
+Assets can be enqueued in a fluent-basis on top of the existing WordPress API
+using the `asset()` helper method. Mantle will read the Webpack manifest and
+automatically determine the URL for the asset.
+
+To get us started, let's enqueue a script from our `entries` directory:
 
 ```php
-asset()->script( 'example-handle', mix( '/js/app.js' ) );
+asset()->script( '/example-entry/index.js' );
 ```
 
-Assets can also be made to load with `async` and/or `defer`:
+Mantle will take it from there and register the asset with WordPress as well as
+all dependencies from `@wordpress/dependency-extraction-webpack-plugin`.
+
+Scripts can also be made to load with `async` and/or `defer`:
 
 ```php
 asset()
-	->script( 'example-handle', mix( '/js/app.js' ) )
+	->script( '/example-entry/index.js' )
 	->async()
 	->defer();
 ```
 
-The location the asset appears can also be controlled:
+Styles can be enqueued in the same way:
 
 ```php
-asset()
-	->script( 'example-handle', mix( '/js/app.js' ) )
-	->header();
+asset()->style( '/example-entry/index.css' );
+```
 
-asset()
-	->script( 'example-footer', mix( '/js/footer.js' ) )
-	->footer();
+### Registering Non-Webpack Assets
+
+Mantle also provides a way to register non-Webpack assets that don't come from
+the Webpack manifest. This is useful for registering assets from plugins or
+other sources.
+
+```php
+asset()->script( 'example-handle', '/path/to/example.js' );
+
+asset()->style( 'example-handle', '/path/to/example.css' );
 ```
 
-You can also specify the asset's version and dependencies:
+The version and other dependencies of the asset can also be specified with the
+`version()` and `dependencies()` method:
 
 ```php
 asset()
-	->style( 'example-styles', mix( '/css/app.css' ) )
+	->script( 'example-handle', '/path/to/example.js' )
 	->version( '1.0' )
-	->dependencies( [ 'unicorn-css' ] );
+	->dependencies( [ 'unicorn-js' ] );
+```
+
+### Asset Conditions
+
+Mantle also provides a way to conditionally enqueue assets. By default, all
+assets registered will be enqueued on all pages. However, you can specify
+conditions for when the asset should be enqueued via Alley's `wp-asset-manager`
+plugin (which Mantle depends on).
+
+```php
+asset()
+	->script( 'example-handle', '/path/to/example.js' )
+	->condition( 'home' );
+```
+
+For more information, checkout the [wp-asset-manager
+documentation](https://github.com/alleyinteractive/wp-asset-manager/#conditions).
+The `Asset_Service_Provider` included with Mantle has a
+`on_am_asset_conditions`` method that can be used to register/manipulate
+conditions.
```

Comment:
Sidebar: I recently realized this plugin is installed even when you are using mantle packages. For example, `mantle-framework/testkit`.

It'd be better if it was bundled inside mantle rather than a standalone plugin in the plugins directory. Unless there is a obvious reason why it needs to be so obvious.

Reply:
agreed, i can work to isolate it